src/renderer/components/ConfigureServer.tsx

```diff
@@ -56,35 +53,68 @@ function ConfigureServer({
         id,
     } = server || {};
 
+    const mounted = useRef(false);
     const [transition, setTransition] = useState<'inFromRight' | 'outToLeft'>();
     const [name, setName] = useState(prevName || '');
     const [url, setUrl] = useState(prevURL || '');
     const [nameError, setNameError] = useState('');
-    const [urlError, setURLError] = useState('');
+    const [urlError, setURLError] = useState<{type: STATUS; value: string}>();
     const [showContent, setShowContent] = useState(false);
     const [waiting, setWaiting] = useState(false);
 
-    const canSave = name && url && !nameError && !urlError;
+    const [validating, setValidating] = useState(false);
+    const validationTimeout = useRef<NodeJS.Timeout>();
+    const [editing, setEditing] = useState(false);
+
+    const canSave = name && url && !nameError && !validating && urlError && urlError.type !== STATUS.ERROR;
 
     useEffect(() => {
         setTransition('inFromRight');
         setShowContent(true);
+        mounted.current = true;
+        return () => {
+            mounted.current = false;
+        };
     }, []);
 
-    const checkProtocolInURL = (checkURL: string): Promise<string> => {
-        if (isValidURL(checkURL)) {
-            return Promise.resolve(checkURL);
+    useEffect(() => {
+        if (editing) {
+            return;
         }
-        return window.desktop.modals.pingDomain(checkURL).
-            then((result: string) => {
-                const newURL = `${result}://${checkURL}`;
-                setUrl(newURL);
-                return newURL;
-            }).
-            catch(() => {
-                console.error(`Could not ping url: ${checkURL}`);
-                return checkURL;
-            });
+
+        fetchValidationResult();
+    }, [editing]);
+
+    const fetchValidationResult = () => {
+        if (!url) {
+            return;
+        }
+
+        setValidating(true);
+        setURLError({
+            type: STATUS.INFO,
+            value: formatMessage({id: 'renderer.components.configureServer.url.validating', defaultMessage: 'Validating...'}),
+        });
+        validateURL(url).then(({validatedURL, serverName, message}) => {
+            if (editing) {
```

Comment:
Given the times we are working with, probably is not going to happen unless you are looking for it, but there is a small issue here. Here the example scenario:
- I write an url
- It starts validating, and for whatever reason, takes ages. Lets call that execution `vURL1`
- I change the url
- It starts validating the second one. `vURL2`.
- Editing now is false.
From this point, two scenarios:
Scenario1:
- vURL1 finishes. `editing` is false, so executes everything. Among other things, rewriting the url the user last posted.
- vURL2 finishes. `editing` is also false, so executes everything. The user is really confused because what he wrote last is back there.
Scenario2:
- vURL2 finishes. `editing` is false, so executes everything. The user is happy with the results.
- vURL1 finishes (it was a long timeout). `editing` is still false, so executes everything. Suddenly the user sees an error and the previous URL. It gets really confused.

To fix this, we can go back to the idea of the timestamp. Actually, we would need both:
- timestamp for several validation running at the same time
- editing so a finishing validation doesn't overwrite what the user is editing right now

0/5 on whether this case is worth handling.

Reply:
Yeah this is why I was originally checking the URL to make sure it matched so that this wouldn't happen, but I understand that it wasn't really working (it seemed like a good idea at the time :P)

The funny thing is there is a case where this could happen, where the URL check is timing out and it's probably going to confuse some people, so maybe we should go back to the timestamp.